core/src/main/scala/io/iteratee/Enumerator.scala

```diff
@@ -160,21 +183,23 @@ final object Enumerator extends EnumeratorInstances {
     xs: IndexedSeq[E],
     min: Int = 0,
     max: Int = Int.MaxValue
-  )(implicit F: Monad[F]): Enumerator[F, E] = new Enumerator[F, E] {
-    private[this] final val limit = math.min(xs.length, max)
-
-    private[this] final def loop[A](pos: Int)(s: Step[F, E, A]): F[Step[F, E, A]] =
-      if (limit > pos) F.flatMap(s.feedEl(xs(pos)))(loop(pos + 1)) else F.pure(s)
-
-    final def apply[A](step: Step[F, E, A]): F[Step[F, E, A]] = loop(math.max(min, 0))(step)
-  }
+  )(implicit F: Applicative[F]): Enumerator[F, E] =
+    new Enumerator[F, E] {
+      final def apply[A](s: Step[F, E, A]): F[Step[F, E, A]] =
+        xs.slice(min, max) match {
+          case is if is.isEmpty => F.pure(s)
+          case IndexedSeq(e) => s.feedEl(e)
```

Comment:
I found it useful to have both `enumVector` and `enumIndexedSeq`, since the latter also works for e.g. ranges.